source/material_model/reactive_fluid_transport.cc

```diff
@@ -0,0 +1,350 @@
+/*
+  Copyright (C) 2023 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/material_model/reactive_fluid_transport.h>
+#include <aspect/simulator_access.h>
+#include <aspect/utilities.h>
+#include <aspect/geometry_model/interface.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/numerics/fe_field_function.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    template <int dim>
+    bool
+    ReactiveFluidTransport<dim>::
+    is_compressible () const
+    {
+      return base_model->is_compressible();
+    }
+
+
+
+    template <int dim>
+    double
+    ReactiveFluidTransport<dim>::
+    reference_darcy_coefficient () const
+    {
+      // 0.01 = 1% melt
+      return reference_permeability * std::pow(0.01,3.0) / eta_f;
+    }
+
+
+
+    template <int dim>
+    void
+    ReactiveFluidTransport<dim>::
+    melt_fractions (const MaterialModel::MaterialModelInputs<dim> &in,
+                    std::vector<double> &melt_fractions) const
+    {
+      for (unsigned int q=0; q<in.temperature.size(); ++q)
+        {
+          const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
+
+          switch (fluid_solid_reaction_scheme)
+            {
+              case no_reaction:
+              {
+                // A fluid-rock reaction model where no reactions occur.
+                // The melt (fluid) fraction at any point is equal
```

Comment:
Perhaps specify this to make a bit more approachable? 
```suggestion
                // The melt (fluid) fraction (volume percent) at any point is equal
```